sys/linux/auparse/auparse.go

```diff
@@ -0,0 +1,359 @@
+// Copyright 2017 Elasticsearch Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auparse
+
+import (
+	"bytes"
+	"fmt"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+//go:generate sh -c "go run mk_audit_msg_types.go && gofmt -w zaudit_msg_types.go"
+//go:generate sh -c "perl mk_audit_syscalls.pl > zaudit_syscalls.go && gofmt -w zaudit_syscalls.go"
+//go:generate perl mk_audit_arches.pl
+
+const (
+	typeToken            = "type="
+	msgToken             = "msg="
+	auditHeaderSeparator = ")"
+)
+
+var (
+	// errInvalidAuditHeader means some part of the audit header was invalid.
+	errInvalidAuditHeader = errors.New("invalid audit message header")
+	// errParseFailure indicates a generic failure to parse.
+	errParseFailure = errors.New("failed to parse audit message")
+)
+
+// AuditMessage represents a single audit message.
+type AuditMessage struct {
+	RecordType AuditMessageType  // Record type from netlink header.
+	Timestamp  time.Time         // Timestamp parsed from payload in netlink message.
+	Sequence   int               // Sequence parsed from payload.
+	RawData    string            // Raw message as a string.
+	Data       map[string]string // The key value pairs parsed from the message.
+	Error      error             // Error that occurred while parsing.
+}
+
+// ToMapStr returns a new map containing the parsed key value pairs, the
+// record_type, @timestamp, and sequence. The parsed key value pairs have
+// a lower precedence than the well-known keys and will knot override them.
```

Comment:
`s/knot/not/`

And describe the `error` key and what conditions under which it will be present in the event.